modules/http/scanner.go

```diff
@@ -446,6 +448,23 @@ func (scan *scan) Grab() *zgrab2.ScanError {
 		bodyText = buf.String()
 	}
 
+	// Application-specific logic for retrying HTTP as HTTPS; if condition matches, return protocol error
+	if scan.scanner.config.FailHTTPToHTTPS && scan.results.Response.StatusCode == 400 && readLen < 1024 && readLen > 24 {
+		// Apache: "You're speaking plain HTTP to an SSL-enabled server port"
+		// NGINX: "The plain HTTP request was sent to HTTPS port"
+		var sliceLen int64 = 128
+		if readLen < sliceLen {
+			sliceLen = readLen
+		}
+		sliceBuf := bodyText[0:sliceLen]
```

Comment:
you can replace with this `sliceBuf := bodyText[:sliceLen]`

Reply:
Thanks @vl4deee11, just committed. I'm very slowly picking up go as you can tell :>